webui/react/src/components/NotesCard.tsx

```diff
@@ -65,26 +99,40 @@ const NotesCard: React.FC<Props> = ({ disabled = false, notes, onSave, style }:
         </Space>
       ) : (
         disabled || (
-          <Tooltip title="Edit">
-            <EditOutlined onClick={editNotes} />
-          </Tooltip>
+          <Space size="middle">
+            <Tooltip title="Edit">
+              <EditOutlined onClick={editNotes} />
+            </Tooltip>
+            {extra}
+          </Space>
         )
       )}
-      headStyle={{ paddingInline: '16px' }}
+      headStyle={{ minHeight: 'fit-content', paddingInline: '16px' }}
       style={{ height: isEditing ? '500px' : '100%', ...style }}
-      title="Notes">
+      title={(
+        <InlineEditor
+          disabled={!onSaveTitle || disabled}
+          focusSignal={noteChangeSignal}
+          style={{ paddingLeft: '5px', paddingRight: '5px' }}
+          value={title}
+          onSave={onSaveTitle}
+        />
+      )}>
       <Spinner spinning={isLoading}>
         <Markdown
           editing={isEditing}
           markdown={isEditing ? editedNotes : notes}
-          onChange={setEditedNotes}
-          onClick={() => { if (notes === '') editNotes(); }}
+          onChange={handleEditedNotes}
+          onClick={(e: React.MouseEvent) => { if (e.detail > 1 || notes === '') editNotes(); }}
         />
       </Spinner>
       <Prompt
         message={(newLocation) => {
+          const isSameExperiment =
+          location.pathname.split('/')[0] === 'experiment' &&
+            newLocation.pathname.startsWith(location.pathname.split('/').slice(0, -1).join('/'));
```

Comment:
**suggestion:** we'll probably use this again in the future for other areas of the app. Make it into a routes utility function?